src/utils/types/object/object.cpp

```diff
@@ -0,0 +1,127 @@
+/***
+ * Maracatronics Robotics
+ * Federal University of Pernambuco (UFPE) at Recife
+ * http://www.maracatronics.com/
+ *
+ * This file is part of Armorial project.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ***/
+
+#include "object.h"
+
+Object::Object() {
+    setInvalid();
+}
+
+Object::~Object() {
+
+}
+
+Position Object::getPosition() {
+    Position retn = _position;
+
+    return retn;
+}
+
+Velocity Object::getVelocity() {
+    Velocity retn = _velocity;
+
+    return retn;
+}
+
+Angle Object::getOrientation() {
+    Angle retn = _orientation;
+
+    return retn;
+}
+
+bool Object::isObjectLoss() {
+    return _lossFilter.checkLoss();
+}
+
+bool Object::isObjectSafe() {
+    return _noiseFilter.checkNoise();
+}
+
+void Object::updateObject(float confidence, Position pos, Angle orientation) {
+    // Update confidence
+    _confidence = confidence;
+
+    // If pos is invalid (robot is not visible in frame)
+    if(pos.isInvalid()) {
```

Comment:
This is the decision tree that runs all the three filters:
- If pos is invalid (robot is not visible in frame):
    - Start loss if not started
        - If is loss, set object as invalid and reset noise
        - If is not loss yet, calls kalman to update object position (using **predict**)
- Else if position is valid
    - Start noise if not initialized and set invalid
        - If is object safe, reset loss and update kalman matrixes with position
        - If is not safe, maintain object as invalid (not safe) and reset loss.